File: Daily/PY/Leetcode1190-反转每对括号间的字符串.py
class Solution:
    def reverseParentheses(self, s: str) -> str:
        cur = ""
        n = len(s)
        stk = []
        for i in range(n):
            if s[i] == '(':
                stk.append(cur)
                cur = ""
            elif s[i] == ')':
                cur = cur[::-1]#字符串反转的方法
                cur = stk.pop(-1) + cur
            else:
                cur += s[i]
        return cur

File: Leetcode/sbgo/Leetcode1190-反转每对括号间的字符串.go
package sbgo

func reverseParentheses(s string) string {
	n := len(s)
	cur := []byte{}
	stk := [][]byte{}
	for i := 0; i < n; i++ {
		if s[i] == '(' {
			stk = append(stk, cur)
			cur = []byte{}
		} else if s[i] == ')' {
			for j, n := 0, len(cur); j < len(cur)/2; j++ {
				cur[j], cur[n-1-j] = cur[n-1-j], cur[j]
			}
			cur = append(stk[len(stk)-1], cur...)
			stk = stk[:len(stk)-1]
		} else {
			cur = append(cur, s[i])
		}
	}
	return string(cur)
}
